packages/appkit/scripts/appkit-version-check.js

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env node
+/* eslint-disable no-console */
+import { existsSync, readFileSync } from 'node:fs'
+import { createRequire } from 'node:module'
+import { dirname, join } from 'node:path'
+import { fileURLToPath } from 'node:url'
+import semver from 'semver'
+
+const require = createRequire(import.meta.url)
+const __dirname = dirname(fileURLToPath(import.meta.url))
+
+/* CONFIG ------------------------------------------------------------------ */
+const REOWN_SCOPE_PREFIX = '@reown/appkit'
+const sections = ['dependencies', 'devDependencies', 'peerDependencies', 'optionalDependencies']
+/* ------------------------------------------------------------------------- */
+
+function findProjectRoot(startPath) {
+  let current = startPath
+
+  // Walk up the directory tree to find the consumer project
+  while (current !== dirname(current)) {
+    const packageJsonPath = join(current, 'package.json')
+
+    if (existsSync(packageJsonPath)) {
+      try {
+        const pkg = JSON.parse(readFileSync(packageJsonPath, 'utf8'))
+
+        // Check if this package.json has @reown/appkit dependencies
+        const hasReownDeps = sections.some(
+          section =>
+            pkg[section] &&
+            Object.keys(pkg[section]).some(dep => dep.startsWith(REOWN_SCOPE_PREFIX))
+        )
+
+        // If it has @reown/appkit deps and it's not the @reown/appkit package itself
+        if (hasReownDeps && pkg.name !== '@reown/appkit') {
+          return { pkg, path: current }
+        }
+      } catch (error) {
+        // Invalid package.json, continue searching
+      }
+    }
+
+    current = dirname(current)
+  }
+
+  return null
+}
+
+function findProjectRootAlternative() {
+  // Alternative approach: use process.env.INIT_CWD (npm) or process.env.PWD
+  const initCwd = process.env.INIT_CWD || process.env.PWD || process.cwd()
+
+  if (initCwd && existsSync(join(initCwd, 'package.json'))) {
+    try {
+      const pkg = JSON.parse(readFileSync(join(initCwd, 'package.json'), 'utf8'))
+      const hasReownDeps = sections.some(
+        section =>
+          pkg[section] && Object.keys(pkg[section]).some(dep => dep.startsWith(REOWN_SCOPE_PREFIX))
+      )
+
+      if (hasReownDeps && pkg.name !== '@reown/appkit') {
+        return { pkg, path: initCwd }
+      }
+    } catch (error) {
+      // Ignore error
+    }
+  }
+
+  return null
+}
+
+// Try multiple approaches to find the consumer project
+let project = findProjectRoot(__dirname) || findProjectRootAlternative()
```

Comment:
can we wrap this code into functions instead of having it at top level? might cause some issues with older node versions

Reply:
also same as above everything should have a try catch